### PR TITLE
Give a useful error message if user gives invalid random seed

### PR DESCRIPTION
**Pull request by jyn514**
I thought this crash was in my code at first because the error was so vague.

### CODE REVIEW

File: src/bin/miri.rs [hunk 1]
```diff
@@ -153,7 +153,14 @@ fn main() {
...
                         panic!("Cannot specify -Zmiri-seed multiple times!");
                     }
-                    let seed_raw = hex::decode(arg.trim_start_matches("-Zmiri-seed=")).unwrap();
+                    let seed_raw = hex::decode(arg.trim_start_matches("-Zmiri-seed="))
+                        .unwrap_or_else(|err| match err {
+                            InvalidHexCharacter { .. } => panic!(
+                                "-Zmiri-seed should only contain valid hex digits [0-9a-fA-F]"
+                            ),
+                            OddLength => panic!("-Zmiri-seed should have an even number of digits"),
+                            _ => panic!("Unknown error decoding -Zmiri-seed as hex"),
```
Comment (bjorn3) [inline src/bin/miri.rs]:
```suggestion
                            err => panic!("Unknown error decoding -Zmiri-seed as hex: {:?}", err),
```

**Comment by RalfJung** [comment]
Thanks a lot! `panic!` is still not the cleanest way to report errors, but it is consistent with the rest of the argument parsing.

However, this fails to build on CI. Can you take care of that?

File: src/bin/miri.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ extern crate rustc_codegen_utils;
 extern crate rustc_interface;
 extern crate syntax;
 
+use hex::FromHexError;
```
Comment (RalfJung) [inline src/bin/miri.rs]: Please keep `std` imports separately. This new import should be below the `rustc` imports, in fact.

Comment (jyn514) [reply src/bin/miri.rs]: Does my new change look right?

File: src/bin/miri.rs [hunk 3]
```diff
@@ -19,6 +19,7 @@ use std::env;
 
 use rustc_interface::interface;
 use rustc::hir::def_id::LOCAL_CRATE;
+use hex::FromHexError;
```
Comment (RalfJung) [inline src/bin/miri.rs]: We usually separate the blocks (std, rustc, other) with an empty line.

Comment (jyn514) [reply src/bin/miri.rs]: I changed it to the style from `src/intptrcast.rs` - `std`, `other`, `rustc`.

Comment (RalfJung) [reply src/bin/miri.rs]: Good catch!

**Comment by RalfJung** [comment]
Thanks!

@bors r+

**Pull request merged**